### PR TITLE
feat(babel): upgrade and set newer browser targets

### PR DESCRIPTION
**Pull request by aymeric-giraudet**
**Summary**

[FX-5053](https://algolia.atlassian.net/browse/FX-3053)

**Result**

- Updated `babel` as well as some plugins
- Changed `browserslist` target
- Added new ESLint rule that outputs errors when using unsupported APIs

### CODE REVIEW

File: .codesandbox/ci.json [hunk 1]
```diff
@@ -18,5 +18,5 @@
     "packages/instantsearch-ui-components",
     "packages/instantsearch-core"
   ],
-  "node": "16"
+  "node": "18"
```
Comment (aymeric-giraudet) [inline .codesandbox/ci.json:21]: This works now ! Node 18 will leave maintenance in April 2025 though

File: babel.config.js [hunk 2]
```diff
@@ -61,46 +61,6 @@ module.exports = (api) => {
         ],
       },
     ],
-    // this plugin is used to test if we need polyfills, not to actually insert them
```
Comment (aymeric-giraudet) [inline babel.config.js:64]: New ESLint rule does that and seems more up to date

File: packages/algoliasearch-helper/documentation-src/content/js/main.js [hunk 3]
```diff
@@ -110,6 +110,7 @@ function activeLinks(sidebarContainer) {
 
 function getPositionsKeyElements(sidebarContainer) {
   const sidebarBBox = sidebarContainer.getBoundingClientRect();
+  // eslint-disable-next-line compat/compat
```
Comment (aymeric-giraudet) [inline packages/algoliasearch-helper/documentation-src/content/js/main.js:113]: Doesn't work in with Firefox < 59 but I guess it's fine

File: packages/vue-instantsearch/rollup.config.js [hunk 4]
```diff
@@ -54,11 +53,6 @@ function outputs(vueVersion) {
     vuePlugin({ compileTemplate: true, css: false }),
     commonjs(),
     json(),
-    buble({
-      transforms: {
-        dangerousForOf: true,
-      },
-    }),
```
Comment (aymeric-giraudet) [inline packages/vue-instantsearch/rollup.config.js:61]: It's unmaintained and not needed anymore

File: .codesandbox/ci.json (hunk 1 above)
Comment (Haroenv) [reply .codesandbox/ci.json:21]: finally! we can revive the vite/astro PRs too then

File: packages/algoliasearch-helper/documentation-src/content/js/main.js (hunk 3 above)
Comment (Haroenv) [reply packages/algoliasearch-helper/documentation-src/content/js/main.js:113]: this is the documentation site, can we disable it from this lint rule instead?

File: packages/vue-instantsearch/rollup.config.js (hunk 4 above)
Comment (Haroenv) [reply packages/vue-instantsearch/rollup.config.js:61]: we used buble back in the day as the output was smaller than babel. Fine to switch to a maintained alternative though

**Review by Haroenv — APPROVED** [review]

File: packages/algolia-experiences/src/render.tsx [hunk 5]
```diff
@@ -257,17 +254,14 @@ function blockToWidget(child: Block, container: HTMLElement): Widget[] {
                 children = ch.children.map(renderChild);
               }
 
-              const attributes = Object.fromEntries(
```
Comment (Haroenv) [inline packages/algolia-experiences/src/render.tsx:260]: is this actually not allowed? from checking caniuse I see older chrome not having this but the other browsers seem ok. Should we have chrome 74 as minimum instead?

File: packages/vue-instantsearch/rollup.config.js (hunk 4 above)
Comment (aymeric-giraudet) [reply packages/vue-instantsearch/rollup.config.js:61]: With/without babel it's a 0.10kb difference only, I forgot to replace it with babel, just did !

File: packages/algolia-experiences/src/render.tsx (hunk 5 above)
Comment (aymeric-giraudet) [reply packages/algolia-experiences/src/render.tsx:260]: Just crunched the numbers and we would lose 1% of users :/

**Review by shaejaz — APPROVED** [review]

**Pull request merged**